test/aggregate.test.js

```diff
@@ -48,20 +48,27 @@ function setupData(db, callback) {
 }
 
 /**
- * Helper function to test operators that only work in MongoDB 3.4 and above (such as some aggregation pipeline operators)
+ * Helper function to test operators that only work in a specific versoin of MongoDB and above (such as some aggregation pipeline operators)
  *
+ * @param {String} semver, `3.4`, specify minimum compatible mongod version
  * @param {Object} ctx, `this`, so that mocha tests can be skipped
  * @param {Function} done
  * @return {Void}
  */
-function onlyTestMongo34(ctx, done) {
+function onlyTestAtOrAbove(semver, ctx, done) {
   start.mongodVersion(function(err, version) {
     if (err) {
       done(err);
       return;
     }
-    var mongo34 = version[0] > 3 || (version[0] === 3 && version[1] >= 4);
-    if (!mongo34) {
+
+    var [major, minor] = semver.split('.').map(function(s) {
```

Comment:
Array destructuring doesn't work in node v4 and v5, which is one of the reason why tests are failing

Reply:
ok, I'll fix it.